### ISSUE TITLE
How to disable xray

### ISSUE THREAD

**Issue by GnatorX** [aws/aws-xray-sdk-go#151]
How do I disable X-ray for non-AWS environments?

**Comment by willarmiros** [comment]
Hi @GnatorX,
While the X-Ray SDK can be disabled in some other languages, it cannot be disabled in Go yet. We always welcome pull requests for new features and it is our goal to move the Go SDK from `rc` to production with feature parity to the other SDKs.

**Comment by victorsmirnov** [comment]
I think the easiest way to disable X-Ray is to provide an Emiter which does nothing. In this case everything is working normally but nothing is send to the daemon. And this is what I wanted from tests.

Please correct me if this approach has any issues.

The code is really simple. I have copy/pasted it from library tests.

```
package xraytest

import (
	"github.com/aws/aws-xray-sdk-go/xray"
	"github.com/rs/zerolog/log"
	"net"
)

type TestEmitter struct{}

func (te *TestEmitter) Emit(seg *xray.Segment) {}

func (te *TestEmitter) RefreshEmitterWithAddress(raddr *net.UDPAddr) {}

func InitXRay() {

	err := xray.Configure(xray.Config{
		Emitter: &TestEmitter{},
	})
	if err != nil {
		log.Fatal().
			Err(err).
			Msg("Failed to initialize X-Ray.")
	}
}
```

Maybe we can also have a configuration option `TestMode bool` and when it is set to true we set TestEmitter and make the library 'silent'. I think this might be common scenario for unit tests or running application locally.

**Comment by cynicaljoy** [comment]
This was addressed in the latest release -- https://github.com/aws/aws-xray-sdk-go/pull/219

**Comment by bhautikpip** [comment]
Released in v1.0.0 so closing this issue. Feel free to re-open for any issues.
